atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/asts/bucketingthings/DefaultSweepAssignedBucketStoreTest.java

```diff
@@ -272,4 +273,42 @@ public void deleteBucketEntryDeletesBucket() {
         store.deleteBucketEntry(bucket);
         assertThat(store.getSweepableBuckets(Set.of(bucket))).isEmpty();
     }
+
+    @Test
+    public void getTimestampRangeRecordThrowsIfRecordNotPresent() {
+        assertThatLoggableExceptionThrownBy(() -> store.getTimestampRangeRecord(1))
+                .isInstanceOf(SafeIllegalStateException.class)
+                .hasLogMessage("No timestamp range record found for bucket identifier")
+                .hasExactlyArgs(SafeArg.of("bucketIdentifier", 1L));
+    }
+
+    @Test
+    public void putTimestampRangeRecordPutsRecord() {
+        TimestampRange timestampRange = TimestampRange.of(1, 2);
+        store.putTimestampRangeRecord(1, timestampRange);
+        assertThat(store.getTimestampRangeRecord(1)).isEqualTo(timestampRange);
+    }
+
+    @Test
+    public void putTimestampRangeRecordFailsIfRecordAlreadyExists() {
+        TimestampRange timestampRange = TimestampRange.of(1, 2);
+        store.putTimestampRangeRecord(1, timestampRange);
+        assertThatThrownBy(() -> store.putTimestampRangeRecord(1, timestampRange))
+                .isInstanceOf(CheckAndSetException.class);
+    }
+
+    @Test
+    @Disabled // TODO(mdaudali): Deletion is not implemented yet
+    public void deleteTimestampRangeRecordDoesNotThrowIfRecordNotPresent() {
+        assertThatCode(() -> store.deleteTimestampRangeRecord(1)).doesNotThrowAnyException();
+    }
+
+    @Test
+    @Disabled // TODO(mdaudali): Deletion is not implemented yet
+    public void deleteTimestampRangeRecordDeletesRecord() {
+        TimestampRange timestampRange = TimestampRange.of(1, 2);
+        store.putTimestampRangeRecord(1, timestampRange);
+        store.deleteTimestampRangeRecord(1);
+        assertThatThrownBy(() -> store.getTimestampRangeRecord(1)).isInstanceOf(SafeIllegalStateException.class);
```

Comment:
when it's time to do this might want to copy the way in which we assert failure on gets here